manifests/repo/apt.pp

```diff
@@ -1,22 +1,60 @@
+# @api private
+#
 # @summary This is a repo class for apt
 #
-# @api private
-class mongodb::repo::apt inherits mongodb::repo {
+# @param ensure
+#   present or absent
+#
+# @param repo_location
+#   Location of the upstream repository
+#
+# @param keyring_location
+#   Location of the upstream keyring
+#
+# @param version
+#   The version of the mongodb repo
+#
+# @param release
+#   Specifies a distribution of the Apt repository.
+#
+# @param repos
+#   Specifies a component of the Apt repository.
+#
+# @param comment
+#   Supplies a comment for adding to the Apt source file.
+#
+class mongodb::repo::apt (
+  Enum['present', 'absent'] $ensure,
+  String[1] $repo_location,
+  String[1] $keyring_location,
+  Optional[String[1]] $release = undef,
+  Optional[String[1]] $repos = undef,
+  Optional[String[1]] $comment = undef,
+) {
   # we try to follow/reproduce the instruction
   # from http://docs.mongodb.org/manual/tutorial/install-mongodb-on-ubuntu/
 
+  assert_private()
+
   include apt
 
-  if($mongodb::repo::ensure == 'present' or $mongodb::repo::ensure == true) {
+  if($ensure == 'present') {
+    # The apt module does not implement gpg --dearmor
```

Comment:
IIRC apt expect file with name ending with `.gpg` to be in the binary OpenPGP format, and files with name ending with `.asc` to use the ASCII armored format.  Downloading the `.asc` file and saving it in a file with the `.asc` extension should work, alternatively saving the `.pub` file with the `.gpg` should also work.  The archive module seems not necessary.

Reply:
Thanks I'll give it a try

Reply:
Tried using
```
    $keyring_file = split($keyring_location, '/')[-1]
    apt::source { 'mongodb':
      location => $repo_location,
      release  => $mongodb::repo::release,
      repos    => $mongodb::repo::repos,
      key      => {
        dir    => '/usr/share/keyrings',
        name   => "mongodb-${keyring_file}",
        source => $keyring_location,
      },
      comment  => $comment,
    }

```
with .asc and .pub. and saving .pub as .gpg
All of them fail with
```
  Notice: /Stage[main]/Apt::Update/Exec[apt_update]/returns: Reading package lists...
  Notice: /Stage[main]/Apt::Update/Exec[apt_update]/returns: W: GPG error: https://repo.mongodb.org/apt/ubuntu focal/mongodb-org/5.0 Release: The following signatures couldn't be verified because the public key is not available: NO_PUBKEY B00A0BD1E2C63C11
  Notice: /Stage[main]/Apt::Update/Exec[apt_update]/returns: E: The repository 'https://repo.mongodb.org/apt/ubuntu focal/mongodb-org/5.0 Release' is not signed.

```

Reply:
Figured it out.
There is a bug in the apt module I think.
When adding signed-by in the sources.list the module does not add the missing '/' when concatenating 'dir' and 'name' ...
Changing 'dir' to '/usr/share/keyrings/' solved that :)